src/Caller.sol

```diff
@@ -85,10 +85,19 @@ contract Caller is EIP712("Caller", "1"), ERC2771Context(address(this)) {
 
     /// @notice Each sender's set of address authorized to make calls on its behalf.
     // slither-disable-next-line naming-convention
-    mapping(address => EnumerableSet.AddressSet) internal _authorized;
+    mapping(address => AddressSetClearable) internal _authorized;
     /// @notice The nonce which needs to be used in the next EIP-712 message signed by the address.
     mapping(address => uint256) public nonce;
 
+    /// @notice A clearable set of addresses.
+    /// @param clears Number of performed clears. Increase to clear.
+    /// @param addressSets The key is the number of performed clears and the value is the set.
+    /// Always use the set under the key equal to the current value of `clears`.
+    struct AddressSetClearable {
```

Comment:
You are completely right that it would be very expensive to remove. :+1: 

Very nice idea with a clear `counter` and new mapping slot for the addresses :+1: